kedro/pipeline/pipeline.py

```diff
@@ -165,6 +178,119 @@ def __init__(
         self._nodes = nodes
         self._topo_sorted_nodes = _topologically_sorted(self.node_dependencies)
 
+    def _transform_pipeline(
```

Comment:
This is exactly the same as it was in modular_pipeline.py, but with (a) the signature changed from
```
def pipeline(
    pipe: Pipeline,
    *,
    inputs: Union[str, Set[str], Dict[str, str]] = None,
    outputs: Union[str, Set[str], Dict[str, str]] = None,
    parameters: Dict[str, str] = None,
    namespace: str = None,
) -> Pipeline:
```
and (b) instances of `pipe` changed to `self` and (c) returns `new_nodes` rather than `Pipeline(new_nodes)`.

Reply:
If you make this public would that be nicer for users?

Reply:
Hmm, I don't think so. Any user who wants to do `pipe2 = pipe1._transform_pipeline(args)` should instead just do `pipe2 = pipeline(pipe1, args)`.